File: src/main/kotlin/org/kite9/diagram/model/style/ConnectionAlignment.java
package org.kite9.diagram.model.style;

public class ConnectionAlignment {

	public static enum Measurement {
		NONE, PERCENTAGE, PIXELS
	}
	
	private final Measurement type;
	private final double amount;
	
	public Measurement getType() {
		return type;
	}

	public double getAmount() {
		return amount;
	}

	public ConnectionAlignment(Measurement type, double amount) {
		super();
		this.type = type;
		this.amount = amount;
	}
	
	public static final ConnectionAlignment NONE = new ConnectionAlignment(Measurement.NONE, 0);
}

File: src/main/kotlin/org/kite9/diagram/model/style/ConnectionAlignment.kt
package org.kite9.diagram.model.style

data class ConnectionAlignment(val type: Measurement, val amount: Double) {

    enum class Measurement {
        NONE, PERCENTAGE, PIXELS
    }

    companion object {

		val NONE = ConnectionAlignment(Measurement.NONE, 0.0)
    }
}